### ISSUE TITLE
[DataGrid] Remove the `ariaV7` experimental flag

### ISSUE THREAD

**Issue by cherniavskii** [mui/mui-x#9874]
TODO:
- [x] Make it a default behavior in v7
- [x] Remove the `ariaV7` experimental flag
- [x] Update failing tests that rely on the previous accessibility attributes
- [x] Migration instructions
- [ ] Add codemod

PR that added the flag: https://github.com/mui/mui-x/pull/9496
